src/DebuggerData.h

```diff
@@ -19,18 +20,25 @@ struct MemoryLayout
 	int mapperSize[4][4];
 };
 
+struct AddressRange {
+    std::optional<uint16_t> start; // a single address when end is not used
+    std::optional<uint16_t> end;   // end point is inclusive
+};
+
+struct Slot {
+    std::optional<int8_t> ps; // primary slot, always 0..3 or empty if it is a mask
+    std::optional<int8_t> ss; // secundary slot, 0..3 or empty when the primary slot is not expanded
+};
+
```

Comment:
> Is it valid for both ps and ss to be empty? What does this mean?

It seems to be a valid situation. A breakpoint may be created with both slot, subslot and segment respectively as `X X X`, `X 1 X` and `X X 1`

> Is it valid for ps to be empty, but ss not?

It can be, although it's a bit weird to create a breakpoint that stops only if ss is set, but ps is not. But if openMSX allows it...